### PR TITLE
♻️ Reword the Rails introduction

### PR DESCRIPTION
**Pull request by purinkle**
Before, the Rails introduction focused on iOS and older framework versions. This was all incorrect as we updated the book to focus on React Native and using the latest Suspenders. We reword the Rails introduction to reflect these changes.

### CODE REVIEW

File: book/rails/introduction.md [hunk 1]
```diff
@@ -1,181 +1,186 @@
...
+Rails. This is because MultiJson is a dependency of ActiveSupport.
 
-In the early days of creating your JSON API, you will likely be changing the
-data returned as well as the data structure on almost a daily basis. While
-communication is both key and challenging on all software development teams, it
-can be especially challenging when working across teams that speak different
-programming languages. While Rails developers and iOS developers alike speak
-"JSON", we found it difficult to ask iOS developers to stay up to date with API
-changes using source code on GitHub.
+For parsing JSON, we chose the Oj gem. To use the Oj gem in your Rails
```
Comment (rhannequin) [inline book/rails/introduction.md:45]: Should this include a link to the [gem's documentation](http://www.ohler.com/oj/doc/index.html)?

File: book/rails/introduction.md [hunk 2]
```diff
@@ -1,181 +1,186 @@
...
+There is no shortage of methods to render a JSON response. We looked into
+[Active Model Serializers][], [RABL][], and using the Rails `.as_json` method.
+We chose [Jbuilder][] for rendering JSON because of its excellent community
+support. The consistency of its view logic to other kinds of Rails views, such
+as Haml, ERB, and Builder, was also a factor.
+
+With Jbuilder, we render the JSON from Rails controllers like any other view. We
+can use partials as with any other Rails view template to compose JSON. Also,
+`cache!` has the same method signature as
+[`ActionView::Helpers::CacheHelper`][]. (It uses `Rails.cache` under the hood.)
```
Comment (rhannequin) [inline book/rails/introduction.md:65]: > If the parenthetical is part of a larger sentence, then the sentence housing the
parenthetical takes care of the punctuation—commas, periods, and anything else will go
outside the parentheses.

> If the parenthetical stands on its own as a sentence without being encased in any larger
sentence, then it is punctuated as its own entity—commas, periods, etc. will go inside the
parentheses.

Here I'd say the parenthesis is part of the previous sentence, as it shares the same subject ("It").

```suggestion
[`ActionView::Helpers::CacheHelper`][] (it uses `Rails.cache` under the hood).
```

File: book/rails/introduction.md [hunk 3]
```diff
@@ -1,181 +1,186 @@
...
+application. To do so, you must maintain the same general JSON data structures
+on the backend for those users.
+
+You will discover new and better ways of structuring your JSON responses. This
+will happen as time progresses and your application grows. Once that happens,
+the easiest way to support old app versions is to release a new API version.
+This will also allow you to add new versions with different JSON structures.
+
+Releasing many versions of an API is outside this book's scope. Ryan Bates has
+an excellent [RailsCast][] on this topic. We will future-proof our API by
```
Comment (rhannequin) [inline book/rails/introduction.md:89]: Should thoughtbot write a book on API versioning? ❤️

File: book/rails/introduction.md [hunk 4]
```diff
@@ -1,181 +1,186 @@
...
+violates the API terms of service, we can disable their key. [This blog post on
+API security][] explains the benefits of API keys.
+
+Humon is an API built for non-commercial purposes. We can create a simple
+permission scheme without many API tokens. But, we want to ensure that only some
+people can query our API. Having zero security would mean anyone could create a
+`curl` request to an endpoint. They could hit our database and cause unknown
+damage.
+
+As a security measure, we need a header of `tb-app-secret` for the `POST users`
```
Comment (rhannequin) [inline book/rails/introduction.md:149]: What do you think of removing the `tb-` prefix, as this is not a thoughtbot-global parameter? Or changing it to something linked to Humon and not thoughtbot?

**Review by rhannequin — APPROVED** [review]
Very nice 👏 
I hope the PDF generation issues could be fixed so that the book keeps getting updated 🙏

**Comment by purinkle** [comment]
I will close this pull request as there are [plans to rewrite the book][] with a different approach. It will be super interesting to see how the next iteration looks.

[plans to rewrite the book]: https://hub.thoughtbot.com/messages/fusion-community-goals

**Pull request closed without merging**